letsencrypt/rootfs/etc/services.d/lets-encrypt/run

```diff
@@ -115,6 +115,14 @@ elif [ "${DNS_PROVIDER}" == "dns-gandi" ]; then
         bashio::log.info "Use Gandi sharing ID"
         echo "dns_gandi_sharing_id = $(bashio::config 'dns.gandi_sharing_id')" >> "/data/dnsapikey"
     fi
+    if bashio::config.exists 'dns.gandi_token'; then
+        bashio::log.info "Use Gandi gandi_token"
+        echo "dns_gandi_token = $(bashio::config 'dns.gandi_token')" >> "/data/dnsapikey"
+    fi
+    if bashio::config.exists 'dns.gandi_api_key'; then
```

Comment:
Should this be an 'else' to avoid strange behavior? (Or an error if both are provided.) It looks like token is preferred according to https://api.gandi.net/docs/authentication/. Alternately maybe these should be reversed to give the token precedence.

Reply:
I guess we could be smarter here, but not sure what the correct smartness is. E.g. the sharing id I think is only valid when an API key is present.

The plug-in just seem to read all of them and forward: https://github.com/obynio/certbot-plugin-gandi/blob/master/certbot_plugin_gandi/main.py#L82-L85. So why not doing it here too :man_shrugging: :sweat_smile:

Reply:
It only forwards one here. Both clauses point to the same target, and if both are set the deprecated one is used because it is last.

Reply:
Not sure what you mean, if all three are set, it forwards all three, at least with this change:

https://github.com/home-assistant/addons/blob/master/letsencrypt/rootfs/etc/services.d/lets-encrypt/run#L114-L125.

Maybe it is easiest if you open a PR what you'd suggest to change, then we can discuss an actual suggestion.